### PR TITLE
Rejigger the vanilla build script to explicitly compile x86 architectures for debug builds only.

### PR DESCRIPTION
**Pull request by twaddington**

### CODE REVIEW

File: simplified-app-vanilla/build.gradle [hunk 1]
```diff
@@ -53,16 +66,6 @@ android {
       applicationIdSuffix ".with_profiles"
     }
   }
-
-  packagingOptions {
-    exclude "META-INF/LICENSE"
```
Comment (twaddington) [inline simplified-app-vanilla/build.gradle:58]: This is excluded by default.

File: simplified-app-vanilla/build.gradle [hunk 2]
```diff
@@ -24,6 +24,17 @@ android {
     setProperty("archivesBaseName", "${versionName}-${versionCode}")
   }
 
+  packagingOptions {
+    doNotStrip 'lib/**/*.so'
```
Comment (twaddington) [inline simplified-app-vanilla/build.gradle:28]: This was added to match the SimplyE build script.

**Comment by io7m** [comment]
I'm unclear what the consequences of this will be. We don't have R1 `x86_64` binaries...

**Comment by twaddington** [comment]
> I'm unclear what the consequences of this will be. 

This shouldn't actually change much of anything 😂 

> We don't have R1 `x86_64` binaries...

You're right, I guess I've never used an `x86_64` emulator image. I removed that.

**Comment by io7m** [comment]
This would work for vanilla builds that won't ever reach the store, but this won't work for SimplyE builds for one reason: Google won't let you publish to the store if you include `x86` binaries but don't include corresponding `x86_64` binaries. I understand what their aim is, but as usual they screwed up the implementation; what if my library _doesn't need_ native binaries for a 64-bit variant of a 32-bit architecture? I've run into this countless times.

**Comment by twaddington** [comment]
> This would work for vanilla builds that won't ever reach the store, but this won't work for SimplyE builds for one reason: Google won't let you publish to the store if you include `x86` binaries but don't include corresponding `x86_64` binaries. I understand what their aim is, but as usual they screwed up the implementation; what if my library _doesn't need_ native binaries for a 64-bit variant of a 32-bit architecture? I've run into this countless times.

**Comment by twaddington** [comment]
oops, hit the wrong button.

**Comment by twaddington** [comment]
> I'm unclear what the consequences of this will be. &mdash; @io7m 

So, right now the vanilla app is building all available architectures by default. This is a bit slow and bloats the APK that is generated. I did two builds, one from `develop` and one from this branch `build-abi` and here's a screenshot comparing the two in the APK analyzer.

[Screen Shot 2020-08-13 at 8.44.02 AM.png](https://user-images.githubusercontent.com/107117/90156408-3fba2d00-dd41-11ea-9a52-3472a55f1de8.png)


The big difference is that the second one, built with an explicit list of ABIs, is 10MB smaller and only includes the architectures we want.

> Google won't let you publish to the store if you include x86 binaries but don't include corresponding x86_64 binaries. I understand what their aim is, but as usual they screwed up the implementation; what if my library doesn't need native binaries for a 64-bit variant of a 32-bit architecture? I've run into this countless times.

That's totally fine because the way this is configured we only build `x86` for debug builds, which are never uploaded to the Play Store. You can see for release builds we still just compile for ARM.

https://github.com/NYPL-Simplified/Simplified-Android-Core/pull/779/files#diff-e1b6dc8da28300b1edb66fa8e7183efcR52

**Comment by twaddington** [comment]
More info about fat binaries and ABI splitting:

- https://developer.android.com/ndk/guides/abis#gc

**Comment by io7m** [comment]
Those poor forgotten MIPS users!  They will never be able to run SimplyE on these:

SGI_Origin3800_300

Let's make these changes.

**Comment by twaddington** [comment]
Thanks Mark! This will make development a smidge faster.

**Pull request merged**